osu.Game/Graphics/DrawableDate.cs

```diff
@@ -75,6 +78,74 @@ private void updateTimeWithReschedule()
 
         private void updateTime() => Text = Format();
 
-        public virtual string TooltipText => string.Format($"{Date:MMMM d, yyyy h:mm tt \"UTC\"z}");
+        public ITooltip GetCustomTooltip() => new DateTooltip();
+
+        public object TooltipContent => Date;
+
+        private class DateTooltip : VisibilityContainer, ITooltip
+        {
+            private readonly OsuSpriteText dateText, timeText;
+            private readonly Box background;
+
+            public DateTooltip()
+            {
+                AutoSizeAxes = Axes.Both;
+                Masking = true;
+                CornerRadius = 5;
+
+                Children = new Drawable[]
+                {
+                    background = new Box
+                    {
+                        RelativeSizeAxes = Axes.Both
+                    },
+                    new FillFlowContainer
+                    {
+                        AutoSizeAxes = Axes.Both,
+                        Direction = FillDirection.Horizontal,
+                        Padding = new MarginPadding(10),
+                        Children = new Drawable[]
+                        {
+                            dateText = new OsuSpriteText
+                            {
+                                Font = OsuFont.GetFont(size: 12, weight: FontWeight.Bold),
+                                Anchor = Anchor.BottomLeft,
+                                Origin = Anchor.BottomLeft,
+                            },
+                            timeText = new OsuSpriteText
+                            {
+                                Font = OsuFont.GetFont(size: 12, weight: FontWeight.Regular),
+                                Anchor = Anchor.BottomLeft,
+                                Origin = Anchor.BottomLeft,
+                            }
+                        }
+                    },
+                };
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                // Temporary colour since it's currently impossible to change it without bugs (see https://github.com/ppy/osu-framework/issues/3231)
+                // If above is fixed, this should use OverlayColourProvider
+                background.Colour = colours.Gray1;
+                timeText.Colour = colours.GreyCyanLighter;
```

Comment:
Since `DrawableDate` is used in more places than just the beatmap set overlay, I think while we can't yet use `OverlayColourProvider` here we should use a neutral shade of grey for now.